ports/webassembly/Makefile

```diff
@@ -15,7 +15,7 @@ INC += -I$(TOP)
 INC += -I$(BUILD)
 
 CFLAGS += -std=c99 -Wall -Werror -Wdouble-promotion -Wfloat-conversion
-CFLAGS += -O3 -DNDEBUG
+CFLAGS += -Os -DNDEBUG
```

Comment:
What's the reason behind this change?

As far as I'm aware, when using `-s ASYNCIFY` (which you may not actually want/need?), you should enable `-O3` otherwise the generated code is too slow.

See https://emscripten.org/docs/porting/asyncify.html#sleeping-yielding-to-the-event-loop, it says:
> It’s very important to optimize (-O3 here) when using Asyncify, as unoptimized builds are very large.

But reading that carefully maybe they mean that you should use some level of optimisation, and maybe `-Os` is indeed better.

If we switch to `-Os` we need to check:
- change in size of generated output
- change in performance/execution speed, possibly measured by the performance benchmark test suite
- both for WASM and non-WASM output, ie `-s WASM=0`)
- both for ASYNCIFY enabled and disabled

Reply:
I was experimenting to see how size of the resulting firmware was changed (see discussion on discord). However, "experimenting" is too grand a word... more like blundering around. This change is the lowest possible priority from my point of view.

Reply:
I had a look at this, here are my results:
```
flags                       firmware.wasm   micropython.js  perf
-O3 -s ASYNCIFY             1342003          212845         0 (baseline)
-O3 -s ASYNCIFY -s WASM=0         -         7064750         -30%
-O3                          367131          196569         +140%
-O3 -s WASM=0                     -         2818260         +30%
-Os -s ASYNCIFY             1135450          213064         +40%
-Os -s ASYNCIFY -s WASM=0         -         6239768         -30%
-Os                          295028          196569         +180%
-Os -s WASM=0                     -         2271358         +30% 
```
The first row is current master.  The second and third columns show firmware size (you need to add them to get the total size).  The fourth column shows the approximate change in performance compared to the master/baseline.

The performance was measured using `run-perfbench.py`.  The error was large, like up to 20%.  But still you could see general trends.

In summary:
- using -Os instead of -O3 makes it a little bit faster in all cases, and smaller output (wasm/js) in all cases
- ASYNCIFY makes the output much bigger, and run much slower
- WASM=0 makes the output much bigger, and run much slower

So I think we should go with this change to `-Os`, it's good all round!

And we should probably add a comment that ASYNCIFY only be used if necessary.

Reply:
Aha... I was meaning to take a look into this (but didn't know how, so thanks for the heads up).

LGTM. :shipit: